File: Java/Hard/WrongAnswer/minimum-skips-to-arrive-at-meeting-on-time_5.java
class Solution {
    public int minSkips(int[] dist, int speed, int hoursBefore) {
        
        int n = dist.length;
        double[][] dp = new double[n][n+1];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j <= n; j++) {
                dp[i][j] = Double.MAX_VALUE;
            }
        }
        dp[0][0] = (double)dist[0] / speed;
        for (int i = 0; i < n-1; i++) {
            for (int j = 0; j <= i; j++) {
                if (dp[i][j] == Double.MAX_VALUE) {
                    continue;
                }
                dp[i+1][j] = Math.min(dp[i+1][j], Math.ceil(dp[i][j]) + (double)dist[i+1] / speed);
                dp[i+1][j+1] = Math.min(dp[i+1][j+1], dp[i][j] + (double)dist[i+1] / speed);
            }
        }
        for (int i = 0; i < n; i++) {
            if (dp[n-1][i] <= hoursBefore) {
                return i;
            }
        }
        return -1;
    }
}

File: Rust/Hard/WrongAnswer/minimum-skips-to-arrive-at-meeting-on-time_5.rs
impl Solution {
    pub fn min_skips(dist: Vec<i32>, speed: i32, hours_before: i32) -> i32 {
        
        let n = dist.len();
        let mut dp = vec![vec![std::i32::MAX; n]; n];
        dp[0][0] = (dist[0] + speed - 1) / speed;
        for i in 1..n {
            dp[i][0] = dp[i - 1][0] + (dist[i] + speed - 1) / speed;
        }
        for i in 1..n {
            for j in 1..=i {
                dp[i][j] = dp[i - 1][j].min(dp[i - 1][j - 1] + (dist[i] + speed - 1) / speed);
                if dist[i] % speed == 0 {
                    dp[i][j] = dp[i][j].min(dp[i - 1][j - 1] + dist[i] / speed - 1);
                }
            }
        }
        for i in 0..n {
            if dp[n - 1][i] <= hours_before {
                return i as i32;
            }
        }
        -1
    }
}